### PR TITLE
raise helpful error message for transforms

### LINKED ISSUE #499: raise helpful error message if pos2vel() is applied without pix2deg()
## Description of the problem

```python
dataset = pm.Dataset('ToyDataset', path='data/ToyDataset')
dataset.download()
dataset.load()
dataset.pos2vel()
```
results in the cryptic error
```
Downloading http://github.com/aeye-lab/pymovements-toy-dataset/zipball/6cb5d663317bf418cec0c9abe1dde5085a8a8ebd/ to data/ToyDataset/downloads/pymovements-toy-dataset.zip
pymovements-toy-dataset.zip: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3.06M/3.06M [00:05<00:00, 640kB/s]Checking integrity of pymovements-toy-dataset.zip
Extracting pymovements-toy-dataset.zip to data/ToyDataset/raw
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:00<00:00, 43.67it/s]  0%|                                                                                                                                                                                     | 0/20 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/user/lab/pymovements/t.py", line 7, in <module>
    dataset.pos2vel()
  File "/home/user/lab/pymovements/src/pymovements/dataset/dataset.py", line 355, in pos2vel
    gaze_df.pos2vel(method=method, **kwargs)
  File "/home/user/lab/pymovements/src/pymovements/gaze/gaze_dataframe.py", line 373, in pos2vel
    self.transform('pos2vel', method=method, **kwargs)
  File "/home/user/lab/pymovements/src/pymovements/gaze/gaze_dataframe.py", line 298, in transform
    [
  File "/home/user/lab/pymovements/src/pymovements/gaze/gaze_dataframe.py", line 299, in <listcomp>
    df.with_columns(transform_method(**kwargs))
  File "/home/user/.local/lib/python3.9/site-packages/polars/dataframe/frame.py", line 7631, in with_columns
    self.lazy()
  File "/home/user/.local/lib/python3.9/site-packages/polars/utils/deprecation.py", line 93, in wrapper
    return function(*args, **kwargs)
  File "/home/user/.local/lib/python3.9/site-packages/polars/lazyframe/frame.py", line 1695, in collect
    return wrap_df(ldf.collect())
exceptions.ColumnNotFoundError: position

Error originated just after this operation:
DF ["text_id", "page_id", "time", "stimuli_x"]; PROJECT */6 COLUMNS; SELECTION: "None"
```

## Description of a solution

If the `GazeDataFrame` has the `pixel` column available, it would be nice to suggest two solutions:

- applying `pix2deg()` beforehand
- taking use of the `position_column` and `velocity_column` keyword arguments from `pos2vel()`

## Minimum acceptance criteria

- [ ] raise error with helpful message somewhere before calling `pl.DataFrame.with_columns()` in  `GazeDataFrame.transform()`

### PR DESCRIPTION
**Pull request by SiQube**
resolves #499 and additionally covers both `pix2deg` and `pos2acc`

### CODE REVIEW

**Review by prassepaul — APPROVED** [review]

File: src/pymovements/gaze/gaze_dataframe.py [hunk 1]
```diff
@@ -266,6 +266,30 @@ def transform(
...
+                        "Neither 'position' is in the columns of the dataframe: "
+                        f'{self.frame.columns} nor is the position column specified.',
+                    )
+            if transform_method.__name__ in {'pix2deg'}:
+                if 'pixel' not in self.frame.columns and 'pixel_column' not in kwargs:
+                    raise pl.exceptions.ColumnNotFoundError(
+                        "Neither 'position' is in the columns of the dataframe: "
+                        f'{self.frame.columns} nor is the pixel column specified. '
+                        'You can specify the pixel column via: '
+                        f'{transform_method.__name__}(pixel_column=name_of_your_pixel_column).',
```
Comment (dkrako) [inline src/pymovements/gaze/gaze_dataframe.py:290]: can you make a string out of `name_of_your_pixel_column`?

File: src/pymovements/gaze/gaze_dataframe.py [hunk 2]
```diff
@@ -266,6 +266,30 @@ def transform(
                 _check_n_components(self.n_components)
                 kwargs['n_components'] = self.n_components
 
+            if transform_method.__name__ in {'pos2vel', 'pos2acc'}:
+                if 'position' not in self.frame.columns and 'position_column' not in kwargs:
+                    if 'pixel' in self.frame.columns:
+                        raise pl.exceptions.ColumnNotFoundError(
+                            "Neither 'position' is in the columns of the dataframe: "
+                            f'{self.frame.columns} nor is the position column specified. '
```
Comment (dkrako) [inline src/pymovements/gaze/gaze_dataframe.py:274]: can you put the list of available columns at the end of the sentence after `[...] specified. `?
Something like: `Available dataframe columns are {self.frame.columns}`

File: src/pymovements/gaze/gaze_dataframe.py [hunk 3]
```diff
@@ -266,6 +266,30 @@ def transform(
...
+                            "Since the dataframe has a 'pixel' column, consider running "
+                            f'pix2deg() before {transform_method.__name__}(). If you want '
+                            'to calculate pixel transformations, you can do so by using '
+                            f"{transform_method.__name__}(position_column='pixel').",
+                        )
+                    raise pl.exceptions.ColumnNotFoundError(
+                        "Neither 'position' is in the columns of the dataframe: "
```
Comment (dkrako) [inline src/pymovements/gaze/gaze_dataframe.py:281]: can you put the list of available columns at the end of the message after `[...] specified. `?
Something like: Available dataframe columns are {self.frame.columns}

File: src/pymovements/gaze/gaze_dataframe.py (hunk 1 above)
Comment (dkrako) [reply src/pymovements/gaze/gaze_dataframe.py:290]: also here too, can you put the list of available columns at the end of the sentence  after `[...] specified. `?
Something like: Available dataframe columns are {self.frame.columns}

**Review by dkrako — CHANGES_REQUESTED** [review]
Thanks a lot for providing these user-friendly error messages!

I just have some cosmetic change requests, that would improve reading flow.

File: src/pymovements/gaze/gaze_dataframe.py (hunk 1 above)
Comment (SiQube) [reply src/pymovements/gaze/gaze_dataframe.py]: done

File: src/pymovements/gaze/gaze_dataframe.py (hunk 2 above)
Comment (SiQube) [reply src/pymovements/gaze/gaze_dataframe.py:274]: done

File: src/pymovements/gaze/gaze_dataframe.py (hunk 3 above)
Comment (SiQube) [reply src/pymovements/gaze/gaze_dataframe.py:282]: done

**Comment by SiQube** [comment]
added the requested changes

**Pull request merged**